src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -38,6 +38,35 @@
 BLACKLISTED_MODS = ['context']
 
 
+class VersionConstraint(object):
+    def __init__(self, resource_type, min_api=None, max_api=None):
+        self._type = resource_type
+        self._min_api = min_api
+        self._max_api = max_api
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def register_cli_argument(self, *args, **kwargs):
+        if supported_api_version(self._type, min_api=self._min_api, max_api=self._max_api):
+            register_cli_argument(*args, **kwargs)
+        else:
+            from azure.cli.core.commands.parameters import ignore_type
+            kwargs = {'arg_type': ignore_type}
+            register_cli_argument(*args, **kwargs)
+
+    def register_extra_cli_argument(self, *args, **kwargs):
+        if supported_api_version(self._type, min_api=self._min_api, max_api=self._max_api):
+            register_extra_cli_argument(*args, **kwargs)
+
+    def cli_command(self, *args, **kwargs):
+        if supported_api_version(self._type, min_api=self._min_api, max_api=self._max_api):
+            cli_command(*args, **kwargs)
```

Comment:
This class looks good!
In `__init__`, you may want to throw an error if both min and max are None.
Or you can leave it as is since `supported_api_version()` with throw anywawy.